jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -25,59 +90,99 @@ protected Action(Tween... tweens) {
         }
     }
 
+    /**
+     * Gets the length (the duration) of the current action.
+     *
+     * @return the length of the action in seconds.
+     */
     @Override
     public double getLength() {
         return length;
     }
 
     /**
-     * Alter the length (duration) of this Action.  This can be used to extend
-     * or truncate an Action.
+     * Alters the length (duration) of this Action.
+     * This can be used to extend or truncate an Action.
      *
      * @param length the desired length (in unscaled seconds, default=0)
      */
     public void setLength(double length) {
         this.length = length;
     }
 
+    /**
+     * Gets the speedup factor applied by the Layer for this action.
+     * The speed factor gets applied on each interpolation by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * Default speed is 1.0, it plays the animation clips at their normal speed.
+     *
+     * @return the speed of frames.
+     */
     public double getSpeed() {
         return speed;
     }
 
+    /**
+     * Sets the speedup factor applied by the Layer running this action.
+     * This factor controls the animation direction, so if the speed is a positive value then the animation would run forward and vice versa.
+     * The speed factor gets applied on each interpolation by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * Default speed is 1.0, it plays the animation clips at their normal speed.
+     *
+     * @param speed the speed of frames.
+     */
     public void setSpeed(double speed) {
         this.speed = speed;
-        if( speed < 0){
-            setForward(false);
-        } else {
-            setForward(true);
-        }
+        setForward(!(speed < 0));
```

Comment:
This change makes the logic less clear. Also, a documentation PR shouldn't change the code.

Reply:
Well, sometimes in order to do some docs, code needs to be simplified.

The original code was : 
```java 
if(speed < 0) {
setForward(false);
}else{
setForward(true);
}
```
So if its simplified then : setForward(!(speed < 0));

But also can be : setForward(speed > 0);

And I think setForward(speed > 0) is the most clear form, in which the animation runs forward if the speed is positive !

Reply:
Not sure why the code had to change for the javadoc to explain what is happening.  Seems like the javadoc describes what the method does whether the old code is used or not.

...javadoc changes should cause 0 regressions.  The only way to guaranteed 0 regressions is to not change the code.

(Strictly speaking, setForward(!(speed < 0)); and setForward(speed > 0); are not equivalent.  setForward(speed >= 0) would be equivalent... and while it _probably_ won't cause a regression, "no code changes" is 100% guaranteed to not cause a regression. ;))

Else we end up with a situation where we can't integrate **javadoc** to some version just because it might change behavior for someone.  Way more maintenance burden than should be required for documentation.

Reply:
Please put the code back the way it was. If you believe the code should be improved, open and issue or a separate PR for that change. A PR should focus on one thing, and in this case that's clearly javadoc.

Reply:
I will revert all the changes in code for now.